Update to reflect new response structure

gliu6 opened the pull request.

gliu6 committed:
Update to reflect new response structure

andhesmi commented on SemanticSearchV2.md:
```diff
@@ -163,19 +163,19 @@ https://api.careerbuilder.com/core/semanticsearch/document
 		- name (string): canonical name
 		- weight (float): estimated weight (0-100)
 		- type (string): skill, job_title, keyword, company, school, location, company_geography
 		- entity_node (map of strings) (occurs only in the following types):
```
shouldn't this be removed entirely?

andhesmi commented on SemanticSearchV2.md:
```diff
@@ -163,19 +163,19 @@ https://api.careerbuilder.com/core/semanticsearch/document
 		- entity_node (map of strings) (occurs only in the following types):
-		    - location; possible fields: address, city, country, region, state, zip
+		   
 		    - experience_level; possible fields: level, minYears, maxYears 
```
this too

gliu6 committed:
add more types

gliu6 committed:
Update SemanticSearchV2.md

Distinguish the PAN responses fields w.rt. whether it is for job or resume only.
Cover all fields in the response.

glpayson commented on SemanticSearchV2.md:
```diff
@@ -264,30 +279,49 @@ The document response is divided into two parts. First is the parsed_input node
+		      "confidence":1,
+		      "id":"11.114"
+		   },
+		   .....
```
Why 5 .'s here and 3 everywhere else?

glpayson commented on SemanticSearchV2.md:
```diff
@@ -264,30 +279,49 @@ The document response is divided into two parts. First is the parsed_input node
+		   "min_years":10,
+		   "max_years":0
+		},
+		"company_geography": [...],
```
Why are we eliding the company_geography? Also +1 for making this plural

glpayson commented on SemanticSearchV2.md:
```diff
@@ -264,30 +279,49 @@ The document response is divided into two parts. First is the parsed_input node
+		"candidate_experience": {
+            		"experience_months": 8,
+            		"experience_months_by_job_category": [
+				{
```
The formatting here is wrong.

glpayson commented on SemanticSearchV2.md:
```diff
@@ -264,30 +279,49 @@ The document response is divided into two parts. First is the parsed_input node
+				...
+	    		]
+		},
+		"geography": [...],
```
Why no example here?

andhesmi replied:
"geography" should be plural too

glpayson commented on SemanticSearchV2.md:
```diff
@@ -264,30 +279,49 @@ The document response is divided into two parts. First is the parsed_input node
+	    		]
+		},
+		"geography": [...],
+		"highest_education_level": "Unknown",
```
Maybe give a better example?

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
-
+	- location: possible fields contain address, city, country, region, state, zip.
+	
+	 *(The following fields are only available when requesting with a job document.)*
```
Should remove 'with':  "when requesting a job document.)*"

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
+	- job_level (string), experience, or seniority level of the job posting.
+	- language_skills (list): contains a list of language skills, each language is a string type.
+	
+	*(The following fields are only available when requesting with a resume document.)*
```
Remove 'with' again.

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
-	- versions (map): relationship strings mapped to version strings
-
-
+	- location: possible fields contain address, city, country, region, state, zip.
```
Should be "possible fields are" since the list is exhaustive.

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
+	- contract_type (string), whether the contract is permanent, temporary, internship, etc.
+	- education_level (string), minimum education level required by the job posting
+	- employment_type (string), whether the job is full-time, part-time, etc.
+	- experience_level: possible fields contain name, level, min_years, max_years.
```
Should also be "possible fields are"

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
+	- employment_type (string), whether the job is full-time, part-time, etc.
+	- experience_level: possible fields contain name, level, min_years, max_years.
+	- job_titles (list): each object contains name, source, confidence, id. 
+	- job_level (string), experience, or seniority level of the job posting.
```
Should be "job_level (string):" not comma.

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
+	- language_skills (list): contains a list of language skills, each language is a string type.
+	
+	*(The following fields are only available when requesting with a resume document.)*
+	- candidate_experience: possible fields contain name, experience_months, experience_months_by_job_category (list) 
```
Should also be "possible fields are"

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
+	- candidate_experience: possible fields contain name, experience_months, experience_months_by_job_category (list) 
+	- geography (list): contains geography objects.
+	- highest_education_level (string), the highest education achieved.
+	- job_type (string), whether the latest job is a fulltime, parttime, etc.
```
"whether the latest job is fulltime,..." lose the 'a'

glpayson commented on SemanticSearchV2.md:
```diff
@@ -264,30 +279,49 @@ The document response is divided into two parts. First is the parsed_input node
-			"custom_keywords": "CustomKeywords"
-		}
+		"location":{  
+		   "address":"Pleasanton, California, Norcross, Georgia",
```
The spacing between colons is inconsistent. Sometimes you do "foo":"bar", sometimes "foo": "bar" and sometimes "foo" : "bar".

gliu6 committed:
Update SemanticSearchV2.md

resolve comments and discussions.

gliu6 committed:
Update SemanticSearchV2.md

Rename fields

gliu6 committed:
Update SemanticSearchV2.md

rephrase.

msundlof commented on SemanticSearchV2.md:
```diff
@@ -163,20 +164,17 @@ https://api.careerbuilder.com/core/semanticsearch/document
 # Document Response
 
-The document response is divided into two parts. First is the parsed_input node which contains information on all extracted keywords and phrases. Next is the extracted_keywords node, which gives the name, type and relationships of all extracted phrases:
+The document response includes the following parts. First is the `parsed_input` node which contains information on all extracted keywords and phrases. Next is the `semantic_keywords` node, which gives the name, type and relationships of all extracted phrases. Also, `job_titles`, `location`, `geography` and etc json objects may also be present. These objects are derived from parsing the job or res ...
```
Instead of "Also, `job_titles`, `location`, `geography` and etc json objects may also be present." I would say "The response may also include additional objects such as, `job_titles`, `location`, and `geography`.

msundlof commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
+	 *(The following fields are only available when requesting a job document.)*
+	- company_geography (list): contains geography objects.
+	- contract_type (string): whether the contract is permanent, temporary, internship, etc.
+	- education_level (string): minimum education level required by the job posting
```
missing a period

gliu6 committed:
Update SemanticSearchV2.md

resolve comments.

msundlof commented on SemanticSearchV2.md:
```diff
@@ -163,20 +164,17 @@ https://api.careerbuilder.com/core/semanticsearch/document
 # Document Response
 
-The document response is divided into two parts. First is the parsed_input node which contains information on all extracted keywords and phrases. Next is the extracted_keywords node, which gives the name, type and relationships of all extracted phrases:
+The document response includes the following parts. First is the `parsed_input` node which contains information on all extracted keywords and phrases. Next is the `semantic_keywords` node, which gives the name, type and relationships of all extracted phrases. Also, the response may also include additional objects such as `job_titles`, `location`, `geographies`. These objects are derived from pars ...
```
Remove "Also" from the beginning. Just "The response..."

gliu6 committed:
Update SemanticSearchV2.md

gliu6 committed:
update job_level value.

gliu6 committed:
update most recent employment level value

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
+	- contract_type (string): whether the contract is permanent, temporary, internship, etc.
+	- education_level (string): minimum education level required by the job posting.
+	- employment_level (string): whether the job is full-time, part-time, etc.
+	- experience_level: possible fields are name, level, min_years, max_years.
```
Shouldn't this be something more like "possible values are..."? It's not the case that there are multiple fields.

gliu6 replied:
I doubt that "possible values" is better. These are fields of experience_level object.
for each of these fields, it makes sense to say like "possible values of min_years field could be integer between 1 and 10."

glpayson commented on SemanticSearchV2.md:
```diff
@@ -185,22 +183,39 @@ The document response is divided into two parts. First is the parsed_input node
+	- employment_level (string): whether the job is full-time, part-time, etc.
+	- experience_level: possible fields are name, level, min_years, max_years.
+	- job_titles (list): each object contains name, source, confidence, id. 
+	- job_level (string): experience, or seniority level of the job posting.
```
I'd maybe say "experience in years"

gliu6 replied:
copied from JPAN documentation, I think this is okay and more robust.

gliu6 committed:
Remove unnecessary filed

remove name from experience_level.

gliu6 committed:
Removed unused field

glpayson reviewed (approved):
This is the official thumb.
